extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/LambdaHttpHandler.java

```diff
@@ -176,6 +177,9 @@ private AwsProxyResponse nettyDispatch(InetSocketAddress clientAddress, AwsProxy
         }
         QuarkusHttpHeaders quarkusHeaders = new QuarkusHttpHeaders();
         quarkusHeaders.setContextObject(Context.class, context);
+        if (request.getRequestContext().getAuthorizer() != null) {
+            quarkusHeaders.add(ApiGatewayAuthorizerContext.class.getName(), request.getRequestContext().getAuthorizer());
```

Comment:
Why are you adding this as a header?  You should be using setContextObject:

quarkusHeaders.setContextObject(ApiGatewayAuthorizerContext.class, request.getRequeestContext().getAuthorizer());

IMO, you should just add a the whole request context instead

quarkusHeaders.setContextObject(AwsProxyRequestContext.class, request.getRequeestContext());

Reply:
That's a much better idea. Thanks!